pkg/controller/upgrade_funcs.go

```diff
@@ -81,6 +81,31 @@ func unmarshalUpgradeObj(ctx context.Context, str string, obj interface{}) error
 	return nil
 }
 
+func walkUpgradeObjs[Obj any](ctx context.Context, objStore objstore.KVStore, dbKey string, workFunc func(ctx context.Context, stm concurrency.STM, buf *Obj) error) error {
```

Comment:
Thanks - that's useful.

Reply:
Yup, trying to reduce boiler plate code.